File: app/src/main/java/com/jnjcomu/edison/activity/SplashActivity.java
package com.jnjcomu.edison.activity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.os.Message;
import android.support.annotation.AnimRes;
import android.support.annotation.NonNull;
import android.support.v7.app.AppCompatActivity;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.ImageView;

import com.jnjcomu.edison.R;

import org.androidannotations.annotations.AfterViews;
import org.androidannotations.annotations.EActivity;
import org.androidannotations.annotations.ViewById;
import org.androidannotations.annotations.res.AnimationRes;

@EActivity(R.layout.activity_splash)
public class SplashActivity extends AppCompatActivity implements Animation.AnimationListener {

    @ViewById(R.id.img_logo)
    ImageView imgLogo;

    @AnimationRes
    Animation dynamicLogo;

    @Override
    public void onBackPressed() {

    }

    @Override
    public void finish() {
        super.finish();
        overridePendingTransition(R.anim.activity_simple_in, R.anim.activity_simple_out);
    }

    @AfterViews
    protected void ready() {
        Handler handler = new Handler();
        handler.postDelayed(() -> imgLogo.startAnimation(getReadyAnimation()), 1000);
    }

    @NonNull
    private Animation getReadyAnimation() {
        dynamicLogo.setAnimationListener(this);
        return dynamicLogo;
    }

    @Override
    public void onAnimationStart(Animation animation) {

    }

    @Override
    public void onAnimationEnd(Animation animation) {
        finish();
        startActivity(new Intent(SplashActivity.this, MainActivity_.class));
    }

    @Override
    public void onAnimationRepeat(Animation animation) {

    }
}

File: app/src/main/java/com/jnjcomu/edison/activity/SplashActivity.kt
package com.jnjcomu.edison.activity

import android.app.AlertDialog
import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.support.v7.app.AppCompatActivity
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import com.jnjcomu.edison.R
import com.jnjcomu.edison.addition.netManager
import com.jnjcomu.edison.api.API
import com.jnjcomu.edison.model.Session
import kotlinx.android.synthetic.main.activity_splash.*
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class SplashActivity : AppCompatActivity(), Animation.AnimationListener {
    private val dynamicLogo: Animation by lazy {
        AnimationUtils.loadAnimation(this, R.anim.dynamic_logo)
    }

    private val readyAnimation: Animation
        get() {
            dynamicLogo.setAnimationListener(this)

            return dynamicLogo
        }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        val handler = Handler()
        handler.postDelayed({ img_logo.startAnimation(readyAnimation) }, 1000)
    }

    override fun onBackPressed() {

    }

    override fun finish() {
        super.finish()

        overridePendingTransition(R.anim.activity_simple_in, R.anim.activity_simple_out)
    }

    override fun onAnimationStart(animation: Animation) {

    }

    override fun onAnimationEnd(animation: Animation) {
        if(!netManager.isConnected()) {
            val builder = AlertDialog.Builder(this)
            builder.setTitle("안내")
            builder.setMessage("네트워크 연결을 확인해주세요.")
            builder.setCancelable(false)
            builder.setPositiveButton("확인", { _, _ ->
                finish()
            })
            builder.show()
        } else {
            val call = API.getApi(this).checkLogin()

            call.enqueue(object : Callback<Session> {
                override fun onResponse(call: Call<Session>?, response: Response<Session>?) {
                    try {
                        if (response?.body()?.login.equals("true")) {
                            startActivity(Intent(applicationContext, MainActivity::class.java))
                            finish()
                        } else {
                            startActivity(Intent(applicationContext, LoginActivity::class.java))
                            finish()
                        }
                    } catch (e: NullPointerException) {
                        startActivity(Intent(applicationContext, LoginActivity::class.java))
                    }
                }

                override fun onFailure(call: Call<Session>?, t: Throwable?) {

                }
            })
        }
    }

    override fun onAnimationRepeat(animation: Animation) {

    }
}
